src/runtime/recipe/handle.ts

```diff
@@ -290,7 +290,7 @@ export class Handle implements Comparable<Handle> {
       typeSet.push({
         // We forced the joined handles to be collections and resolve their type first,
         // so that we can pull out their collection type here.
-        type: new TupleType(this.joinedHandles.map(h => (h.type as CollectionType<Type>).collectionType.referenceTo())).collectionOf(),
+        type: new TupleType(this.joinedHandles.map(h => (h.type as CollectionType<Type>).collectionType)).collectionOf(),
```

Comment:
what if someone actually does make a Tuple of references, will this still work?

Reply:
Yes, it will, it's just that we're not automatically wrapping everything into references. So if you join a collection of entities, you get entities, if you join a collection of references, you get references.